src/Testcontainers/Configurations/Modules/Databases/AzuriteTestcontainerConfiguration.cs

```diff
@@ -0,0 +1,279 @@
+    /// <param name="image">The Docker image.</param>
+    protected AzuriteTestcontainerConfiguration(string image)
+    {
+      this.Image = image;
+      this.Environments = new Dictionary<string, string>();
+      this.OutputConsumer = Consume.DoNotConsumeStdoutAndStderr();
+    }
+
+    /// <summary>
+    /// Gets the Docker image.
+    /// </summary>
+    [PublicAPI]
+    public string Image { get; }
+
+    /// <summary>
+    /// Gets or sets the host Blob port.
+    /// </summary>
+    /// <remarks>
+    /// Corresponds to the default port of the hosted service.
+    /// </remarks>
+    [PublicAPI]
+    public int BlobPort { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether Blob service should run standalone.
+    /// </summary>
+    /// <remarks>
+    /// Default value is false.
+    /// </remarks>
+    [PublicAPI]
+    public bool RunBlobOnly
+    {
+      get => this.runBlobOnly;
+      set
+      {
+        this.runBlobOnly = value;
+        if (value)
+        {
+          this.RunQueueOnly = this.RunTableOnly = false;
+        }
+      }
+    }
+
+    /// <summary>
+    /// Gets or sets the host Queue port.
+    /// </summary>
+    /// <remarks>
+    /// Corresponds to the default port of the hosted service.
+    /// </remarks>
+    [PublicAPI]
+    public int QueuePort { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether Queue service should run standalone.
+    /// </summary>
+    /// <remarks>
+    /// Default value is false.
+    /// </remarks>
+    [PublicAPI]
+    public bool RunQueueOnly
+    {
+      get => this.runQueueOnly;
+      set
+      {
+        this.runQueueOnly = value;
+        if (value)
+        {
+          this.RunBlobOnly = this.RunTableOnly = false;
+        }
+      }
+    }
+
+    /// <summary>
+    /// Gets or sets the host Table port.
+    /// </summary>
+    /// <remarks>
+    /// Corresponds to the default port of the hosted service.
+    /// </remarks>
+    [PublicAPI]
+    public int TablePort { get; set; }
+
+    /// <summary>
+    /// Gets or sets a value indicating whether Table service should run standalone.
+    /// </summary>
+    /// <remarks>
+    /// Default value is false.
+    /// </remarks>
+    [PublicAPI]
+    public bool RunTableOnly
+    {
+      get => this.runTableOnly;
+      set
+      {
+        this.runTableOnly = value;
+        if (value)
+        {
+          this.RunBlobOnly = this.RunQueueOnly = false;
+        }
+      }
+    }
+
+    /// <summary>
+    /// Gets a value indicating whether all Azurite service will run.
+    /// </summary>
+    [PublicAPI]
+    public bool RunAllServices => !this.RunBlobOnly && !this.RunQueueOnly && !this.RunTableOnly;
+
+    /// <summary>
+    /// Gets or sets workspace location path.
+    /// </summary>
+    /// <remarks>
+    /// Corresponds to the default workspace location of the hosted service.
+    /// </remarks>
+    [PublicAPI]
+    public string Location
```

Comment:
`[NotNull]`

Reply:
`[NotNull]` means that returned value cannot be null which is not true in this case because default value is null. I change implementation of this and add `[CanBeNull]` in [bddb8be](https://github.com/testcontainers/testcontainers-dotnet/pull/555/commits/bddb8be4dc91b51a89a704799bbe006d784e4f8c)

Reply:
I think my idea was to use `DefaultLocation` instead. I try to avoid `null` as much as I can, but I don't remember anymore.